homeassistant/components/switch/tellduslive.py

```diff
@@ -0,0 +1,77 @@
+"""
+homeassistant.components.switch.tellduslive
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Tellstick switches using Tellstick Net and
+the Telldus Live online service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.tellduslive/
+
+"""
+import logging
+
+from tellcore.constants import (
+    TELLSTICK_TURNON, TELLSTICK_TURNOFF)
+
+from homeassistant.const import (STATE_ON, STATE_OFF, STATE_UNKNOWN)
+from homeassistant.components import tellduslive
+from homeassistant.helpers.entity import ToggleEntity
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = ['tellduslive']
+TELLSTICK_UNKNOWN = 0
+STATES = {TELLSTICK_UNKNOWN: STATE_UNKNOWN,
+          TELLSTICK_TURNON:  STATE_ON,
+          TELLSTICK_TURNOFF: STATE_OFF}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Find and return Tellstick switches. """
+    switches = tellduslive.NETWORK.get_switches()
+    add_devices([TelldusLiveSwitch(switch["name"],
+                                   switch["id"],
+                                   STATES[switch["state"]])
+                 for switch in switches if switch["type"] == "device"])
+
+
+class TelldusLiveSwitch(ToggleEntity):
+    """ Represents a Tellstick switch. """
+
+    def __init__(self, name, switch_id, state):
+        self._name = name
+        self._id = switch_id
+        self._state = state
+
+    @property
+    def should_poll(self):
+        """ Tells Home Assistant to poll this entity. """
+        return True
```

Comment:
If the switch turning on/off works you can just have it return `False` for now and keep the optimistic approach of predicting state.